### PR TITLE
Prevent the verification of a verified number and address

### LINKED ISSUE #2217: Verify Phone number flow is trigger even when user is signup with same backup key and phone number.
**Frequency:** 100%
**App version:** Staging build v1.5.2
**Repro on:** Samsung Galaxy Note 5 (7.0), Samsung Galaxy s5 (6.0.1)  

**Pre-condition:** Users should signup with the same phone number backup key. 

**Repro Steps:** 
1) Launch the app.   
2) Enter a phone number. 
3) Enter the security PIN. 
4) Click on Restore backup key. 
5) Paste the same backup key for the phone number. 
6) Tap on the Restore button. 
7) Skip the contact flow. 
8) Observed Verify phone number flow is trigger. 

**Investigation:** 
The user did not get the three verification and after a few seconds “Congratulations, You are verified” message is shown. 
**Impact:** User may get frustrated as user has already verified that phone and used a backup of that phone number. 
**Current Behavior:** Verify Phone number flow is a trigger
**Expected Behavior:** Verify Phone number flow should not be trigger if the user has already verified that phone number. 
**Attachment:** [Android_VerifiedPhoneNumber.mp4 ](https://drive.google.com/file/d/1amZfEnvXTXjC7LF9q3BzC39LjiWg7OXE/view?usp=sharing)(Bug observed at 1: 01 sec)

### PR DESCRIPTION
**Pull request by Pedro-vk**
### Description

_Prevent the verification of a verified number and address_

### Tested

_Tested on Android device_

### Related issues

- Fixes #2217

### CODE REVIEW

File: packages/mobile/src/identity/verification.ts [hunk 1]
```diff
@@ -63,6 +64,30 @@ export interface AttestationCode {
...
+
+  const status: AttestationsStatus = yield call(
+    getAttestationsStatus,
+    attestationsWrapper,
+    account,
+    e164Number
+  )
+
+  if (status.isVerified) {
+    yield put(setVerificationStatus(VerificationStatus.Done))
```
Comment (jmrossy) [inline packages/mobile/src/identity/verification.ts]: Let's not set this. It's meant to track status updates through the verification process but the user won't be doing that process.

File: packages/mobile/src/import/saga.ts [hunk 2]
```diff
@@ -76,6 +77,9 @@ export function* importBackupPhraseSaga({ phrase, useEmptyWallet }: ImportBackup
     yield put(refreshAllBalances())
     navigate(Screens.ImportContacts)
     yield put(importBackupPhraseSuccess())
+
+    // Check if the account was verified
+    yield call(checkVerification)
```
Comment (jmrossy) [inline packages/mobile/src/import/saga.ts]: This must be done before the navigate call above or users could click skip before the saga is done

**Review by jmrossy — COMMENTED** [review]
Overall looks good! Just some small things to fix

File: packages/mobile/src/import/saga.ts [hunk 3]
```diff
@@ -74,7 +75,12 @@ export function* importBackupPhraseSaga({ phrase, useEmptyWallet }: ImportBackup
     // Set redeem invite complete so user isn't brought back into nux flow
     yield put(redeemInviteSuccess())
     yield put(refreshAllBalances())
-    navigate(Screens.VerificationEducationScreen)
+
+    // Check if the account was verified
+    const isVerified = yield call(checkVerification)
+
+    navigate(isVerified ? Screens.ExchangeHomeScreen : Screens.VerificationEducationScreen)
```
Comment (jmrossy) [inline packages/mobile/src/import/saga.ts]: You should navigate to WalletHome, not ExchangeHome

Comment (Pedro-vk) [reply packages/mobile/src/import/saga.ts]: Done!

**Review by jmrossy — APPROVED** [review]

**Pull request merged**